src/System.Private.CoreLib/src/System/Enum.CoreCLR.cs

```diff
@@ -15,13 +16,13 @@ public abstract partial class Enum
         private static extern void GetEnumValuesAndNames(RuntimeTypeHandle enumType, ObjectHandleOnStack values, ObjectHandleOnStack names, bool getNames);
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        public override extern bool Equals(object obj);
+        public override extern bool Equals(object? obj);
 
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern object InternalBoxEnum(RuntimeType enumType, long value);
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern int InternalCompareTo(object o1, object o2);
+        private static extern int InternalCompareTo(object o1, object? o2);
```

Comment:
While it may not strictly be necessary, it's strange that one of the inputs is defined to be non-null and the other nullable.  Let's just make them both be `object?` for consistency.

Reply:
I think we will have plenty of APIs where we have a mix; I don't think we should be sacrificing accuracy for consistency.

Reply:
> I think we will have plenty of APIs where we have a mix; I don't think we should be sacrificing accuracy for consistency.

I thought InternalCompareTo allowed both to be null.  But I just looked and it doesn't.  In which case the annotation is correct but the names are bad.  These should be named `thisRef` and `target` or something like that to reflect that the arguments are not interchangeable.

Reply:
> These should be named thisRef and target or something like that to reflect that the arguments are not interchangeable.

Agreed. There are likely a number of internal helper APIs with parameters that could do with similar name fixes.

Reply:
@stephentoub yes i checked pinvoked method, only target can be null. 
Question: are you suggesting me to rename these variables? i thougth we shouldn't do such refactoring ...

Reply:
> Question: are you suggesting me to rename these variables? i thought we shouldn't do such refactoring

Doesn't need to be in this PR.